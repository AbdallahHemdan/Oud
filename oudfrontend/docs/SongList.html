<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SongList</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SongList</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SongList<span class="signature">(recieved, track, tracks, pause, resume, addToQueue, clickedItemId)</span><span class="type-signature"></span></h2>
        
            <div class="class-description">this is a component that renders the list of songs in playlists, albums, likedSongs or a "loading" animation if the songs are not recieved</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="SongList"><span class="type-signature"></span>new SongList<span class="signature">(recieved, track, tracks, pause, resume, addToQueue, clickedItemId)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>recieved</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">the number of songs in the playlists, albums, likedSongs</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>track</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last">song object
                <h6>Properties</h6>
                

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">track id</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>name</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">track title</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>albumId</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the id of the track's album</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>type</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the type of music that the track belongs to</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>audioUrl</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the URL of the track</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>artist._id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the Id of the singer</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>artist.name</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">name of the singer</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>artist.type</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the type of music that the artist produces</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>artist.image</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">Image of the artist</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>track.artists</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;artist></span>


            
            </td>

            

            

            <td class="description last">the artists that produced the track</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>tracks</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;track></span>


            
            </td>

            

            

            <td class="description last">array of songs to be rendered</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>pause</code></td>
            

            <td class="type">
            
                
<span class="param-type">func</span>


            
            </td>

            

            

            <td class="description last">pauses the player</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>resume</code></td>
            

            <td class="type">
            
                
<span class="param-type">func</span>


            
            </td>

            

            

            <td class="description last">resumes the player</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>addToQueue</code></td>
            

            <td class="type">
            
                
<span class="param-type">func</span>


            
            </td>

            

            

            <td class="description last">adds array of tracks to queue</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>clickedItemId</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">used to mark all songs as unclicked</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li><a href="mailto:ahmedwa1999@gmail.com">Ahmed Walid</a></li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_commonComponents_songList.jsx.html">components/commonComponents/songList.jsx</a>, <a href="components_commonComponents_songList.jsx.html#line44">line 44</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        


    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="componentWillReceiveProps"><span class="type-signature"></span>componentWillReceiveProps<span class="signature">(nextProps)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    if the recieved props is changed it sets the state to the new props
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>nextProps</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_commonComponents_songList.jsx.html">components/commonComponents/songList.jsx</a>, <a href="components_commonComponents_songList.jsx.html#line66">line 66</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="handleClick"><span class="type-signature"></span>handleClick<span class="signature">(id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    it sets the state tothe new ID so it is sent to all the songs
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the id of the calling song</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_commonComponents_songList.jsx.html">components/commonComponents/songList.jsx</a>, <a href="components_commonComponents_songList.jsx.html#line59">line 59</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="handlePlay"><span class="type-signature"></span>handlePlay<span class="signature">(id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    if the calling song is playing it pauses/resums the playerotherwise, it calls handleQueue
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">sttring</span>


            
            </td>

            

            

            <td class="description last">the id of the calling song</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_commonComponents_songList.jsx.html">components/commonComponents/songList.jsx</a>, <a href="components_commonComponents_songList.jsx.html#line92">line 92</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="handleQueue"><span class="type-signature"></span>handleQueue<span class="signature">(id)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    it adds all the songs after the calling song to the queue
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">the id of the calling song</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_commonComponents_songList.jsx.html">components/commonComponents/songList.jsx</a>, <a href="components_commonComponents_songList.jsx.html#line76">line 76</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AccountOverview.html">AccountOverview</a></li><li><a href="addToPlaylist.html">addToPlaylist</a></li><li><a href="Album.html">Album</a></li><li><a href="Albums.html">Albums</a></li><li><a href="ChangePassword.html">ChangePassword</a></li><li><a href="CreatePlaylist.html">CreatePlaylist</a></li><li><a href="EditProfile.html">EditProfile</a></li><li><a href="Facebook.html">Facebook</a></li><li><a href="FollowCard.html">FollowCard</a></li><li><a href="Followers.html">Followers</a></li><li><a href="Following.html">Following</a></li><li><a href="ForgotPassword.html">ForgotPassword</a></li><li><a href="GenreCard.html">GenreCard</a></li><li><a href="HeaderBody.html">HeaderBody</a></li><li><a href="HeaderBodyTop.html">HeaderBodyTop</a></li><li><a href="Home.html">Home</a></li><li><a href="Library.html">Library</a></li><li><a href="LikedSongs.html">LikedSongs</a></li><li><a href="MainBrand.html">MainBrand</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="MusicCard.html">MusicCard</a></li><li><a href="MusicItem.html">MusicItem</a></li><li><a href="Navbar.html">Navbar</a></li><li><a href="Player.html">Player</a></li><li><a href="PlayingBarRight.html">PlayingBarRight</a></li><li><a href="Playlist.html">Playlist</a></li><li><a href="Playlists.html">Playlists</a></li><li><a href="PublicPlaylists.html">PublicPlaylists</a></li><li><a href="Queue.html">Queue</a></li><li><a href="ResetPassword.html">ResetPassword</a></li><li><a href="SideBar.html">SideBar</a></li><li><a href="Sidebar_.html">Sidebar</a></li><li><a href="SignIn.html">SignIn</a></li><li><a href="Signup.html">Signup</a></li><li><a href="SocialIcons.html">SocialIcons</a></li><li><a href="Song.html">Song</a></li><li><a href="SongList.html">SongList</a></li><li><a href="Track.html">Track</a></li><li><a href="UpperContainer.html">UpperContainer</a></li><li><a href="UserExperienceForm.html">UserExperienceForm</a></li><li><a href="WebPlayer.html">WebPlayer</a></li></ul><h3>Global</h3><ul><li><a href="global.html#addToQueue">addToQueue</a></li><li><a href="global.html#BeforeLogin">BeforeLogin</a></li><li><a href="global.html#CardList">CardList</a></li><li><a href="global.html#checkPassword">checkPassword</a></li><li><a href="global.html#EditCountry">EditCountry</a></li><li><a href="global.html#EditGendr">EditGendr</a></li><li><a href="global.html#EditProfileTextElement">EditProfileTextElement</a></li><li><a href="global.html#FeedBack">FeedBack</a></li><li><a href="global.html#fetchCategoriesUrl">fetchCategoriesUrl</a></li><li><a href="global.html#Footer">Footer</a></li><li><a href="global.html#formValid">formValid</a></li><li><a href="global.html#HeaderBodyBottom">HeaderBodyBottom</a></li><li><a href="global.html#HelpUs">HelpUs</a></li><li><a href="global.html#HiddenSideBar">HiddenSideBar</a></li><li><a href="global.html#ImageContainer">ImageContainer</a></li><li><a href="global.html#LowerContainer">LowerContainer</a></li><li><a href="global.html#MainContainer">MainContainer</a></li><li><a href="global.html#Overview">Overview</a></li><li><a href="global.html#PageContainer">PageContainer</a></li><li><a href="global.html#pause">pause</a></li><li><a href="global.html#PlayingBarCenter">PlayingBarCenter</a></li><li><a href="global.html#PlayingBarLeft">PlayingBarLeft</a></li><li><a href="global.html#Profile">Profile</a></li><li><a href="global.html#ProfileElement">ProfileElement</a></li><li><a href="global.html#ProfileInfo">ProfileInfo</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#resume">resume</a></li><li><a href="global.html#SideBarElement">SideBarElement</a></li><li><a href="global.html#SignOutEverywhere">SignOutEverywhere</a></li><li><a href="global.html#SortableTrackContainer">SortableTrackContainer</a></li><li><a href="global.html#ThankYou">ThankYou</a></li><li><a href="global.html#TrackContainer">TrackContainer</a></li><li><a href="global.html#YourPlan">YourPlan</a></li><li>{JSX}</li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Wed Apr 22 2020 21:05:29 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>