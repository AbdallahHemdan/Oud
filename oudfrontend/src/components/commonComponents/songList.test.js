import React from 'react'
import SongList from './songList';
import Enzyme, { shallow } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import checkPropTypes from 'check-prop-types'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const fullProps = {
    recieved: true,
    tracks: [
        {
            "id": "19",
            "name": "إنسان",
            "artists": [
                {
                    "id": "5",
                    "name": "حمزة نمرة",
                    "type": "shaaby",
                    "image": "string"
                },
                {
                    "id": "2",
                    "name": "Sardeena",
                    "type": "Shaaby",
                    "image": "string"
                }
            ],
            "album":{
                "name":"ddd"
            },
            "albumId": "1",
            "type": "Jazz",
            "audioUrl": "www.Facebook.com"
        },
        {
            "id": "20",
            "name": "سلاح التلميذ",
            "artists": [
                {
                    "id": "3",
                    "name": "ًWegz el wench",
                    "type": "Trap",
                    "image": "string"
                },
                {
                    "id": "4",
                    "name": "Amr Diab",
                    "type": "farafeery",
                    "image": "string"
                }
            ],
            "album":{
                "name":"ddd"
            },
            "albumId": "1",
            "type": "Trap",
            "audioUrl": "www.facebook.com"
        }],
    pause: jest.fn(),
    resume: jest.fn(),
    addToQueue: jest.fn(),
    clickedItemId: '0',
    
    webPlayer:{current:{playContext:jest.fn()}}
}
const setup = (props = {}) => {
    return shallow(<SongList {...props} />);
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
}
describe('SongList Component', () => {
    describe('testing SongList Component with recieved equal false', () => {
        let component;
        let props = { recieved: false }
        beforeEach(() => {
            component = setup(props);
        })
        it('renders songsList correctly when recived = false', () => {
            const wrapper = findByTestAttr(component, 'songsList');
            expect(wrapper.length).toBe(1);
        });
        it("doesn't render songElement when recived = false", () => {
            const wrapper = findByTestAttr(component, 'songElement');
            expect(wrapper.length).toBe(0);
        });
        it("doesn't render songElement when recived = false", () => {
            const wrapper = findByTestAttr(component, 'loading');
            expect(wrapper.length).toBe(1);
        });
    });

    describe('testing SongList Component without props', () => {
        let component;
        beforeEach(() => {
            component = setup();
        })
        it('renders songsList correctly when without props', () => {
            const wrapper = findByTestAttr(component, 'songsList');
            expect(wrapper.length).toBe(1);
        });
        it("doesn't render songElement when without props", () => {
            const wrapper = findByTestAttr(component, 'songElement');
            expect(wrapper.length).toBe(0);
        });
        it("doesn't render songElement when without props", () => {
            const wrapper = findByTestAttr(component, 'loading');
            expect(wrapper.length).toBe(1);
        });
    });

    describe('testing SongList Component with recieved equal true and tracks is empty', () => {
        let component;
        let props = { recieved: true, tracks: [] }
        beforeEach(() => {
            component = setup(props);
        })
        it('render songsList correctly when recived = true and tracks is empty', () => {
            const wrapper = findByTestAttr(component, 'songsList');
            expect(wrapper.length).toBe(1);
        });
        it("doesn't render songElement when recived = true and tracks is empty", () => {
            const wrapper = findByTestAttr(component, 'songElement');
            expect(wrapper.length).toBe(0);
        });
        it("doesn't render songElement when without props", () => {
            const wrapper = findByTestAttr(component, 'loading');
            expect(wrapper.length).toBe(0);
        });
    });

    describe('testing SongList Component with recieved equal true and tracks not empty', () => {
        let component;
        beforeEach(() => {
            component = setup(fullProps);
        })
        it('render songsList correctly when recived = true and tracks not empty', () => {
            const wrapper = findByTestAttr(component, 'songsList');
            expect(wrapper.length).toBe(1);
        });
        it("doesn't render songElement when recived = true and tracks is not empty", () => {
            const wrapper = findByTestAttr(component, 'songElement');
            expect(wrapper.length).toBe(2);
        });
        it("doesn't render songElement when without props", () => {
            const wrapper = findByTestAttr(component, 'loading');
            expect(wrapper.length).toBe(0);
        });
    });
    describe('checking propTypes', ()=>{
        const props10 ={recieved:true, tracks:{}}
        const props11 ={recieved:true, tracks:[]}
        const props01 ={recieved:4, tracks:[]}
        const props00 ={recieved:4, tracks:{}}
        //testing with full props
        it('should not throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { ...fullProps }, 'prop', SongList.name);
            expect(result).toBeUndefined();
        })

        //testing pause function by sending invalid data
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { pause: '1' }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { pause: 1 }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { pause: true }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })

        //testing resume function by sending invalid data
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { resume: '1' }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { resume: 1 }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { resume: true }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })

        //testing addToQueue function by sending invalid data
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { addToQueue: '1' }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { addToQueue: 1 }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { addToQueue: true }, 'prop', SongList.name);
            expect(result).toBeDefined();
        })

        //testing clicked Item Id by sending false data
        it('should not throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, { clickedItemId: 1 }, 'prop', SongList.name);
            expect(result).toBeDefined();
        });


        it('should not throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, props11, 'prop', SongList.name);
            expect(result).toBeUndefined();
        });
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, props10, 'prop', SongList.name);
            expect(result).toBeDefined();
        });
        it('should not throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, props01, 'prop', SongList.name);
            expect(result).toBeDefined();
        });
        it('should throw a warning', () => {
            const result = checkPropTypes(SongList.propTypes, props00, 'prop', SongList.name);
            expect(result).toBeDefined();
        });
    });
    describe('calling functions', ()=>{
        let component;
        beforeEach(()=>{
            component = setup(fullProps)
        });
        it('calling handleClick ', ()=>{
            component.setState({clickedItemId:'0'})
            component.instance().handleClick('2');
            expect(component.state().clickedItemId).toBe('2');
        });
        it('change the recieved props', () => {
            component.setState({clickedItemId:'0'})
            component.setProps({clickedItemId:'2'})
            expect(component.state().clickedItemId).toBe('2');
        })
        it('change the recieved props', () => {
            component.setState({clickedItemId:'0'})
            component.setProps({clickedItemId:'0'})
            expect(component.state().clickedItemId).toBe('0');
        })
        it('calling handlePlay ', ()=>{
            const comp = setup({...fullProps, contextType:'album'})
            comp.setState({playing:true})
            comp.instance().handlePlay('19');
            expect(comp.state().playing).toBe(false);
        });
        it('calling handlePlay ', ()=>{
            component.setState({playing:true})
            component.instance().handlePlay('19');
            expect(component.state().playing).toBe(false);
        });
        it('calling handlePlay ', ()=>{
            const comp = setup({...fullProps, contextType:'playlist'})
            comp.setState({playing:false})
            comp.instance().handlePlay('19');
            expect(comp.state().playing).toBe(true);
        });
    });
    describe('snapshot test', () => {
        it('renders correctly', () => {
            const tree = renderer
                .create(<SongList {...fullProps} />)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
