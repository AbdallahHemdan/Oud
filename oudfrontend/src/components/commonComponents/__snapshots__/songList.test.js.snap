// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongList Component snapshot test renders correctly 1`] = `
<div
  className="col-xs-8 col-md-6 col-lg-8 col-xl-8"
  data-testid="songsList"
>
  <button
    className="song row"
    data-testid="song"
    id="song"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="songIcon col-1"
    >
      <button
        className="songButton goldText"
        data-testid="playButton"
        onClick={[Function]}
      >
        <img
          alt="play music icon"
          data-testid="playButtonImage"
          height="20"
          src="musicIcon.png"
          width="18"
        />
      </button>
    </div>
    <div
      className="songInfo col-8"
    >
      <span
        className="whiteText"
        data-testid="songName"
      >
        إنسان
      </span>
      <p
        data-testid="aristsNames"
      >
        <span>
          <span>
            <button
              className="playlistAnchor songButton"
              data-testid="artistName"
              onClick={[Function]}
            />
            <span
              className="gray-text"
              data-testid="comma"
            >
               
              •
               
            </span>
          </span>
          <span>
            <button
              className="playlistAnchor songButton"
              data-testid="artistName"
              onClick={[Function]}
            />
            <span
              className="gray-text"
              data-testid="comma"
            >
               
              •
               
            </span>
          </span>
        </span>
        <button
          className="playlistAnchor songButton"
          data-testid="albumName"
          onClick={[Function]}
        >
          ddd
        </button>
      </p>
    </div>
    <div
      className="col-1"
    >
      <div
        className="dropdowns"
        data-testid="dropdown"
      >
        <button
          className="songButton hide"
          data-testid="dropdownButton"
          id="songDropdownButton"
          onClick={[Function]}
        >
          <h3
            className="whiteText"
          >
            ...
          </h3>
        </button>
        <div
          className="dropdownContent hide"
          data-testid="dropdownList"
          id="dropdownContent"
        >
          <button
            className="SongDropdownItem songButton"
            data-testid="saveSong"
            onClick={[Function]}
          >
            Save to your Liked Songs
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="addToQueue"
            onClick={[Function]}
          >
            Add to Queue
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="addToPlaylist"
            onClick={[Function]}
          >
            Add to Playlist
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="copy"
            onClick={[Function]}
          >
            Copy Song Link
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-2"
    >
      <p
        className="whiteText"
        data-testid="songTime"
      >
        NaN:NaN
      </p>
    </div>
  </button>
  <button
    className="song row"
    data-testid="song"
    id="song"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="songIcon col-1"
    >
      <button
        className="songButton goldText"
        data-testid="playButton"
        onClick={[Function]}
      >
        <img
          alt="play music icon"
          data-testid="playButtonImage"
          height="20"
          src="musicIcon.png"
          width="18"
        />
      </button>
    </div>
    <div
      className="songInfo col-8"
    >
      <span
        className="whiteText"
        data-testid="songName"
      >
        سلاح التلميذ
      </span>
      <p
        data-testid="aristsNames"
      >
        <span>
          <span>
            <button
              className="playlistAnchor songButton"
              data-testid="artistName"
              onClick={[Function]}
            />
            <span
              className="gray-text"
              data-testid="comma"
            >
               
              •
               
            </span>
          </span>
          <span>
            <button
              className="playlistAnchor songButton"
              data-testid="artistName"
              onClick={[Function]}
            />
            <span
              className="gray-text"
              data-testid="comma"
            >
               
              •
               
            </span>
          </span>
        </span>
        <button
          className="playlistAnchor songButton"
          data-testid="albumName"
          onClick={[Function]}
        >
          ddd
        </button>
      </p>
    </div>
    <div
      className="col-1"
    >
      <div
        className="dropdowns"
        data-testid="dropdown"
      >
        <button
          className="songButton hide"
          data-testid="dropdownButton"
          id="songDropdownButton"
          onClick={[Function]}
        >
          <h3
            className="whiteText"
          >
            ...
          </h3>
        </button>
        <div
          className="dropdownContent hide"
          data-testid="dropdownList"
          id="dropdownContent"
        >
          <button
            className="SongDropdownItem songButton"
            data-testid="saveSong"
            onClick={[Function]}
          >
            Save to your Liked Songs
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="addToQueue"
            onClick={[Function]}
          >
            Add to Queue
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="addToPlaylist"
            onClick={[Function]}
          >
            Add to Playlist
          </button>
          <button
            className="SongDropdownItem songButton"
            data-testid="copy"
            onClick={[Function]}
          >
            Copy Song Link
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-2"
    >
      <p
        className="whiteText"
        data-testid="songTime"
      >
        NaN:NaN
      </p>
    </div>
  </button>
</div>
`;
